references/optical_flow/train.py

```diff
@@ -269,13 +293,19 @@ def main(args):
         # Note: we don't sync the SmoothedValues across processes, so the printed metrics are just those of rank 0
         print(f"Epoch {current_epoch} done. ", logger)
 
-        if args.rank == 0:
-            # TODO: Also save the optimizer and scheduler
-            torch.save(model.state_dict(), Path(args.output_dir) / f"{args.name}_{current_epoch}.pth")
-            torch.save(model.state_dict(), Path(args.output_dir) / f"{args.name}.pth")
+        if not args.distributed or args.rank == 0:
+            checkpoint = {
+                "model": model.state_dict(),
```

Comment:
Let's load and save the `model_without_ddp` instead of the model. This way, we would be able to seemlessly load a model that was trained with DDP in a non-DDP script, and vice-versa. Take a look at how we do that in the classification training reference - it should only take ~3 lines of code to be added. Happy to chat more about it if you need clarifications.